middle_end/flambda2/simplify/env/continuation_uses.ml

```diff
@@ -108,8 +108,11 @@ let get_arg_types_by_use_id_for_invariant_params arity l =
     (fun arg_maps t ->
       if not
            (Misc.Stdlib.List.is_prefix ~equal:Flambda_kind.equal
-              (Flambda_arity.to_list arity)
-              ~of_:(Flambda_arity.to_list t.arity))
+              (Flambda_arity.to_list arity
+              |> List.map Flambda_kind.With_subkind.kind)
+              ~of_:
+                (Flambda_arity.to_list t.arity
+                |> List.map Flambda_kind.With_subkind.kind))
```

Comment:
Likewise, this could become:
```ocaml
          (Misc.Stdlib.List.is_prefix ~equal:Flambda_kind.With_subkind.equal_ignoring_subkind
             (Flambda_arity.to_list arity)
             ~of_:(Flambda_arity.to_lsit t.arity))
```